build-setup/vscode/using_vscode.md

```diff
@@ -0,0 +1,232 @@
+            "problemMatcher": [
+                "$gcc"
+            ],
+            "dependsOn": [
+                "Build ml-cpp lib"
+            ],
+            "options": {
+                "env": {
+                    "CPP_SRC_HOME": "${env:CPP_SRC_HOME}"
+                }
+            }              
+        },
+        {
+            "label": "Build ml-cpp maths tests",
+            "type": "shell",
+            "command": "cd ${workspaceFolder}/lib/maths/unittest && make ML_DEBUG=1 -j6",
+            "problemMatcher": [
+                "$gcc"
+            ],
+            "dependsOn": [
+                "Build ml-cpp lib"
+            ],
+            "options": {
+                "env": {
+                    "CPP_SRC_HOME": "${env:CPP_SRC_HOME}"
+                }
+            }                       
+        },
+        {
+            "label": "Build ml-cpp bin",
+            "type": "shell",
+            "command": "cd ${workspaceFolder}/bin && make ML_DEBUG=1 -j6",
+            "problemMatcher": [
+                "$gcc"
+            ],
+            "dependsOn": [
+                "Build ml-cpp lib"
+            ],
+            "options": {
+                "env": {
+                    "CPP_SRC_HOME": "${env:CPP_SRC_HOME}"
+                }
+            }               
+        }
+    ]
+}
+```
+
+## Debugging
+
+To use [debugging](https://code.visualstudio.com/docs/editor/debugging), you need to specify the binary and the
+arguments in a json file. You can use `preLaunchTask` to specify the user tasks that should be executed before the start
+(e.g. building project).
+
+Here is a example of the `launch.json` file:
+
+```json
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "data_frame_analyzer",
+            "type": "cppdbg",
+            "request": "launch",
+            "program": "${workspaceFolder}/build/distribution/platform/linux-x86_64/bin/data_frame_analyzer",
+            "args": ["--input", "OnlineNewsPopularity_small.csv",
+                "--config", "es_regression_configuration_small.json", 
+                "--output", "output.txt"],
+            "stopAtEntry": false,
+            "cwd": "${workspaceFolder}",
+            "environment": [],
+            "externalConsole": false,
+            "MIMode": "gdb",
+            "setupCommands": [
+                {
+                    "description": "Enable pretty-printing for gdb",
+                    "text": "-enable-pretty-printing",
+                    "ignoreFailures": true
+                }
+            ],
+            "preLaunchTask": "Build ml-cpp bin"
+        },
+        {
+            "name": "api unit test",
+            "type": "cppdbg",
+            "request": "launch",
+            "program": "${workspaceFolder}/lib/api/unittest/ml_test",
+            "args": [
+                "--run_test= CInferenceModelMetadataTest/testJsonSchema"
```

Comment:
This is a useful example but we shouldn't have it in the setup instructions as the user would be fooled into thinking all the tests have run.

Reply:
The `args` option is now documented at the end of the file